std/array.d

```diff
@@ -587,11 +592,16 @@ if (isInputRange!Values && isInputRange!Keys)
 {
     import std.algorithm.iteration : filter, map;
     import std.range : enumerate;
+    import std.range.primitives : autodecodeStrings;
 
     auto r = "abcde".enumerate.filter!(a => a.index == 2);
     auto a = assocArray(r.map!(a => a.value), r.map!(a => a.index));
     assert(is(typeof(a) == size_t[dchar]));
-    assert(a == [dchar('c'): size_t(2)]);
+    static if (autodecodeStrings)
+        alias achar = dchar;
+    else
+        alias achar = immutable(char);
+    assert(a == [achar('c'): size_t(2)]);
```

Comment:
How about this instead (shorter and less DRY):
```d
assert(a == [ElementType!string('c'): size_t(2)]);
```

Reply:
It's a good idea for user code, but for a test case I think it's better to see what the actual type is rather than a proxy for it.